### PR TITLE
Fix configuration formatting of zamg sensor and weather component

### PR DESCRIPTION
**Pull request by cgtobi**
**Description:**
Fix configuration formatting of zamg sensor and weather component.
Related to #6385.

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/sensor.zamg.markdown [hunk 1]
```diff
@@ -25,24 +25,46 @@ sensor:
...
-  - **humidity**: Humidity
-  - **wind_speed**: Wind speed
-  - **wind_bearing**: Wind bearing
-  - **wind_max_speed**: Top wind speed
-  - **wind_max_bearing**: Top wind bearing
-  - **sun_last_hour**: Sun last hour percentage
-  - **temperature**: Temperature
-  - **precipitation**: Precipitation
-  - **dewpoint**: Dew point
+{% configuration %}
```
Comment (klaasnicolaas) [inline source/_components/sensor.zamg.markdown:28]: Good work 👍, but you still miss the type and / or default variables

File: source/_components/weather.zamg.markdown [hunk 2]
```diff
@@ -27,12 +27,20 @@ weather:
   - platform: zamg
 ```
 
-Configuration variables:
-
-- **station_id** (*Optional*): The ID number for a supported ZAMG station.
-- **name** (*Optional*): A name for the weather platform.
-- **latitude** (*Optional*): Latitude coordinate to monitor weather of (required if **longitude** is specified). Defaults to coordinates defined in your `configuration.yaml` file.
-- **longitude** (*Optional*): Longitude coordinate to monitor weather of (required if **latitude** is specified). Defaults to coordinates defined in your `configuration.yaml` file.
+{% configuration %}
```
Comment (klaasnicolaas) [inline source/_components/weather.zamg.markdown:30]: Same

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/sensor.yweather.markdown [hunk 3]
```diff
@@ -44,12 +44,15 @@ sensor:
 woeid:
   required: false
   description: See above.
+  type: string
 forecast:
   required: false
   description: Day of forecast. The default is the current day to display conditions.
+  type: integer
 name:
```
Comment (klaasnicolaas) [inline source/_components/sensor.yweather.markdown:54]: Add default variable and remove "Defaults to `Yweather`." from the description

File: source/_components/sensor.yweather.markdown [hunk 4]
```diff
@@ -44,12 +44,15 @@ sensor:
 woeid:
   required: false
   description: See above.
+  type: string
 forecast:
```
Comment (klaasnicolaas) [inline source/_components/sensor.yweather.markdown:49]: Also missing the default variable = 0

File: source/_components/sensor.zamg.markdown [hunk 5]
```diff
@@ -25,24 +25,55 @@ sensor:
...
+station_id:
+  required: false
+  description: The ID number for a supported ZAMG station.
+  type: string
```
Comment (klaasnicolaas) [inline source/_components/sensor.zamg.markdown:32]: Actually this should be an integer but mad enough I see in the python code that there is a string for it 🤔

File: source/_components/sensor.zamg.markdown [hunk 6]
```diff
@@ -25,24 +25,55 @@ sensor:
...
+name:
+  required: false
+  description: Additional name for the sensors. Defaults to platform name.
+  default: zamg
+  type: string
+latitude:
+  required: false
+  description: "Latitude coordinate to monitor weather of (required if **longitude** is specified)."
+  default: "Defaults to coordinates defined in your `configuration.yaml` file."
+  type: string
```
Comment (klaasnicolaas) [inline source/_components/sensor.zamg.markdown]: Must be a float

File: source/_components/sensor.zamg.markdown [hunk 7]
```diff
@@ -25,24 +25,55 @@ sensor:
...
+longitude:
+  required: false
+  description: "Longitude coordinate to monitor weather of (required if **latitude** is specified)."
+  default: "Defaults to coordinates defined in your `configuration.yaml` file."
+  type: string
```
Comment (klaasnicolaas) [inline source/_components/sensor.zamg.markdown]: Must be a float

File: source/_components/weather.zamg.markdown [hunk 8]
```diff
@@ -27,12 +27,26 @@ weather:
...
+  type: string
+name:
+  required: false
+  description: A name for the weather platform.
+  type: string
+latitude:
+  required: false
+  description: "Latitude coordinate to monitor weather of (required if **longitude** is specified)."
+  default: "Defaults to coordinates defined in your `configuration.yaml` file."
+  type: string
```
Comment (klaasnicolaas) [inline source/_components/weather.zamg.markdown]: Float

File: source/_components/weather.zamg.markdown [hunk 9]
```diff
@@ -27,12 +27,26 @@ weather:
...
+longitude:
+  required: false
+  description: "Longitude coordinate to monitor weather of (required if **latitude** is specified)."
+  default: "Defaults to coordinates defined in your `configuration.yaml` file."
+  type: string
```
Comment (klaasnicolaas) [inline source/_components/weather.zamg.markdown]: Float

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Review by frenck — CHANGES_REQUESTED** [review]
Update PR as per review @klaasnicolaas.

File: source/_components/sensor.zamg.markdown (hunk 5 above)
Comment (cgtobi) [reply source/_components/sensor.zamg.markdown:32]: The website returns a CSV, so everything is a string.

See: https://github.com/home-assistant/home-assistant/blob/07972c84a15740a8de47883749f7eb9301bcdc89/homeassistant/components/sensor/zamg.py#L178

Comment (klaasnicolaas) [reply source/_components/sensor.zamg.markdown:32]: Do not say that it should be an integer but it could be 😉

Comment (cgtobi) [reply source/_components/sensor.zamg.markdown:32]: I am with you on that.

File: source/_components/sensor.yweather.markdown (hunk 4 above)
Comment (klaasnicolaas) [reply source/_components/sensor.yweather.markdown:49]: Is solved now

**Review by klaasnicolaas — APPROVED** [review]

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @cgtobi!

Happy Hacktoberfest 🎉 

../Frenck

**Pull request merged**